components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,101 @@
+public class HybridConfigPersistenceManagerTest extends ConfigPersistenceManagerTest {
+
+    private JDBCConfigPersistenceManager jdbcConfigPersistenceManager;
+    private RegistryConfigPersistenceManager registryConfigPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        configPersistenceManager = new HybridConfigPersistenceManager();
+        jdbcConfigPersistenceManager = new JDBCConfigPersistenceManager();
+        registryConfigPersistenceManager = new RegistryConfigPersistenceManager();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+
+        deletePolicyCombiningAlgorithmInDatabase();
+        registryConfigPersistenceManager.deleteGlobalPolicyAlgorithm();
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromRegistry(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        registryConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        String policyAlgorithmFromRegistry = configPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromRegistry, policyAlgorithm);
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromDatabase(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        jdbcConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        String policyAlgorithmFromDatabase = configPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromDatabase, policyAlgorithm);
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testUpdateGlobalPolicyAlgorithmInRegistry(String policyAlgorithm) throws Exception {
+
+        registryConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(DENY_OVERRIDES);
+        configPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        // Verify that the global policy combining algorithm value was deleted from the registry.
+        assertFalse(registryConfigPersistenceManager.isGlobalPolicyAlgorithmExists());
+
+        String policyAlgorithmFromDatabase = configPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromDatabase, policyAlgorithm);
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testUpdateGlobalPolicyAlgorithmInDatabase(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        configPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(DENY_OVERRIDES);
+        configPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        String policyAlgorithmFromDatabase = configPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromDatabase, policyAlgorithm);
```

Comment:
Better to add `assertFalse(registryConfigPersistenceManager.isGlobalPolicyAlgorithmExists());` to here as well..
Without that we cant make sure that data pulled from the database impl.

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5905/commits/fe59d644b547388f83603a9ac1a0e90effde1ec4